### ISSUE TITLE
Add option to sync map with url

### ISSUE THREAD

**Issue by silvester-pari** [EOX-A/EOxElements#132]
This could be enabled with an attribute, e.g. `link-to-url` (there might be better naming closer to vanilla OL).

Some ideas what could be linked:
- center/zoom/rotation (configurable, one or multiple)
- center in projection of map (e.g. lat/lon by default or mercator)
- visible layers?
- drawn shapes?
- ...

**Comment by RobertOrthofer** [comment]
the standard ol `link`-interaction manipulates `window.location`, which is self-contained inside the web component. Apparently we do have access to the parent-location via `window.parent.location` or `window.top.location`, so we would have to keep this in sync. I am not sure if it's advisable to try to manipulate the parent applications `window.history`. 

Connecting to our meeting today, maybe being able to access the current map config (which should be in-sync with the map anyways) would be sufficient for most use cases? View options like zoom and center and layer definitions and visibility would be defined, the entire map should be reproducable with this information.

**Comment by silvester-pari** [comment]
I don't believe there should be an issue with accessing `window` from the web component 🤔 ; but I agree the map config should be always up to date, maybe we can "simply" link the map config with the window query?

**Comment by RobertOrthofer** [comment]
It's no issue to do it from a technical point of view, I rather thought about from a logical perspective, as there can be multiple instances of the EOx map in an application at the same time

I think there are multiple options for this feature:

 - work with hooks for zoom/center and let the parent-application handle everything
 - have a simple link to the URL (opt-in), each instance of the EOx map element that is linked to the URL reacts to URL-changes (which can be dangerous and limiting, but on the other hand could also be seen as a feature for some applications)
 - have some kind of prefix for each EOx-map, so when 2 EOx maps are mounted in the application, each has their own representation in the URL

**Comment by silvester-pari** [comment]
I like a combination of 2 & 3, maybe an opt-in attribute/property e.g. `sync-url` that, if set, syncs that specific map with the url; if there are two, we could pass map ids, e.g. `sync-url="map-a"`. What do you think?
Possible namings:
- `sync-url`
- `url-sync`
- `link-to-url`
- `url-link`
- ...?
